docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,83 @@
+# Rapid releases and moving towards a monorepo
+
+* Status: proposed
+* Deciders: ?
+* Date: ?
+
+## Context and Problem Statement
+
+There are several issues with the current application-services release process:
+  - Our release schedule is relatively infrequent and consumer apps often take a long time to take our releases.  This means that there's a long turnaround time between when code gets merged to our main branch and when it gets released to a nightly build where it can be used and tested.  This conflicts with the rapid release model used by Firefox Desktop, where code changes are frequently released in nightlies.
+  - Releases require too much developer involvement.  Normal releases take a couple hours, during which a developer needs to monitor the progress and perform several manual steps along the way.  If the unreleased code has breaking changes, then the developer also needs to merge the PR for the consumer application(s) after the release, which is not ideal since that developer is often not the one who wrote the PR.  Worse, if the dev who caused the breaking changes forgot to create a PR, they need to create that PR now, leading to multiple devs coordinating for the release.  Releases that are not from main also require extra work: creating a branch for that version, backporting the fix, making a patch release, etc.
+  - The mobile teams are moving towards a weekly release cycle, which may lead to us needing to do more releases for older versions in order to uplift fixes.
+
+At the same time, Mozilla is in the process of moving towards a single monorepo.  In a monorepo, the application-services code will be immediately consumed by all applications as soon as it's merged.  This is drastically different than our current workflow, where code sits in main for long periods of time before being released, then sits in a release for a long period of time before being taken by all consumers.
```

Comment:
I thought the plan was that iOS, android, and everything else, would eventually end up in moz-central -- although it's going to be a long time and we haven't decided on the specific steps.

Reply:
I like the idea of linking to the proposal document, but hesitate because it's a private google doc at this point.  What do others think, link or no link?

Reply:
I vote linking it is OK, this ADR is mostly relevant to Mozilla employees anyways 🤷